Author: Jeffrey Bastian <jbastian@redhat.com>
Date:   Tue Nov 11 16:59:31 2014 -0600

    replace .byte with actual rdrand and rdseed
    
    gas will not generate DWARF data for instructions entered as .byte
    sequences, and binutils in RHEL7 knows about these instructions
    so just use them directly

diff --git a/rdrand_asm.S b/rdrand_asm.S
index 9918363d4a34..f2cfa7bfe5e1 100644
--- a/rdrand_asm.S
+++ b/rdrand_asm.S
@@ -38,7 +38,7 @@ ENTRY(x86_rdrand_bytes)
 1:
 	mov	$RDRAND_RETRY_LIMIT, %ecx
 2:
-	.byte	0x48,0x0f,0xc7,0xf2	/* rdrand %rdx */
+	rdrand	%rdx
 	jnc	3f
 	mov	%rdx, (%rdi)
 	add	$8, %rdi
@@ -60,7 +60,7 @@ ENTRY(x86_rdseed_or_rdrand_bytes)
 1:
 	mov	$RDRAND_RETRY_LIMIT, %r10d
 2:
-	.byte	0x48,0x0f,0xc7,0xf8	/* rdseed %rax */
+	rdseed	%rax
 	jnc	3f
 	mov	%rax, (%rdi)
 	add	$8, %rdi
@@ -71,7 +71,7 @@ ENTRY(x86_rdseed_or_rdrand_bytes)
 	sub	%r9d, (%rcx)
 	ret
 3:
-	.byte	0x48,0x0f,0xc7,0xf0	/* rdrand %rax */
+	rdrand	%rax
 	jnc	5f
 	mov	%rax, (%rdx)
 	add	$8, %rdx
@@ -106,7 +106,7 @@ ENTRY(x86_rdrand_bytes)
 1:
 	mov	$RDRAND_RETRY_LIMIT, %ecx
 2:
-	.byte	0x0f,0xc7,0xf2		/* rdrand %edx */
+	rdrand	%edx
 	jnc	3f
 	mov	%edx, (%edi)
 	add	$4, %edi
@@ -142,7 +142,7 @@ ENTRY(x86_rdseed_or_rdrand_bytes)
 1:
 	mov	$RDRAND_RETRY_LIMIT, %ecx
 2:
-	.byte	0x0f,0xc7,0xf8		/* rdseed %eax */
+	rdseed	%eax
 	jnc	3f
 	mov	%eax, (%edi)
 	add	$4, %edi
@@ -160,7 +160,7 @@ ENTRY(x86_rdseed_or_rdrand_bytes)
 	pop	%ebp
 	ret
 3:
-	.byte	0x0f,0xc7,0xf0		/* rdrand %eax */
+	rdrand	%eax
 	jnc	5f
 	mov	%eax, (%edx)
 	add	$4, %edx
